### DISCUSSION TITLE
service-cloud-connector | DW API call from SalesCloud is giving error "NO DNS NAME".

### DISCUSSION THREAD

**Discussion by PremSolanki** [SalesforceCommerceCloud/sfcc-ci#540] [Team Posts]
When We are making a DW api call from Sales-cloud, Sales-cloud server is giving error "No DNS name...", I checked the my DW server certificate that is not valid but after uploading of new SSH certificate its not working.
Is there any solution from Salecloud side ?
Error : 
![image](https://user-images.githubusercontent.com/22399834/56354562-b4369780-61f1-11e9-8af1-dfb29ed804dd.png)

**Comment by pravngaur** [comment]
Its a java security exception while making a https call using the ssl
certificate.

The certificate itself is not correct, please validate the steps that you
took for creating the certificate. Have a look at :
http://www.littlebigextra.com/how-to-fix-javax-net-ssl-sslhandshakeexception-java-security-cert-certificateexception-no-subject-alternative-names-present/

Also, since you are running this on local sandbox, it would be fine if you
run it on http only and use certificates on higher sandboxes. Just a
suggestion if you don't get the error resolved. But definitely certificate
issue, follow the link above to check the validity of the certificate.

**Comment by shayunBBD** [comment]
Hello @PremSolanki , I have the same issue from ServiceCloud.
How did you fix it? How did you check your DW certificate?

Thank you

**Comment by PremSolanki** [comment]
Hi @shayunBBD  , 
This issue was because of Dw certificate.
First try to create Commerce cloud URL structure like :  instance name-realm-customer name.demandware.net
if your URL structure is like :   instance name.realm.customer name.demandware.net

This will allow HTTPS on your sandbox if this wont work than you URL You need to raise a support ticket to Demandware. They will import the certificate for the realm.

Thanks
Prem
